src/esw/imu_driver.py

```diff
@@ -63,30 +119,42 @@ def main():
             rospy.logerr("incomplete msg")
             continue
 
+        # rotate the quaternion and mag vector by 90 degrees about the Z axis to convert it to ENU frame
+        offset_quat = quaternion_about_axis(np.pi / 2, [0, 0, 1])
+        enu_orientation = quaternion_multiply(offset_quat, orientation_data)
+        R = rotation_matrix(np.pi / 2, [0, 0, 1])
+        enu_mag_vec = R @ np.array(mag_data)
+
         # fill in all sensor messages, setting timestamps of each message to right now,
         # and setting the reference frame of all messages to IMU frame
+        header = Header(stamp=rospy.Time.now(), frame_id=imu_frame)
+        mag_msg = MagneticField(
+            header=header,
+            magnetic_field=Vector3(*enu_mag_vec),
+        )
         imu_msg = ImuAndMag(
-            header=Header(stamp=rospy.Time.now(), frame_id=imu_frame),
+            header=header,
             imu=Imu(
-                header=Header(stamp=rospy.Time.now(), frame_id=imu_frame),
-                orientation=Quaternion(*orientation_data),
+                header=header,
+                orientation=Quaternion(*enu_orientation),
                 linear_acceleration=Vector3(*accel_data),
                 angular_velocity=Vector3(*gyro_data),
+                orientation_covariance=orientation_covariance.flatten().tolist(),
```

Comment:
Yeah, basically the message wants a flat 1D array of covariances, but I want to be able to put a 2D matrix of covariances in the config file. And I use numpy arrays instead of a list of lists since numpy just has a flatten() function to convert to what the message wants

Reply:
fixed to not flatten every time